### ISSUE TITLE
CHMC from A Family of MCMC Methods on Implicitly Defined Manifolds

### ISSUE THREAD

**Issue by krzysztofrusek** [blackjax-devs/blackjax#643]
I am starting this issue is to discuss my work on the algorithm from [A Family of MCMC Methods on Implicitly Defined Manifolds](https://www.cs.toronto.edu/~mbrubake/projects/AISTATS12.pdf) by @mbrubake as discussed in in #538 .

In the #644  I will migrate from [CHMC](https://github.com/krzysztofrusek/jax_chmc) to blackjax.

### TODO

- [x] RATTLE integrator
- [x] Implicit manifold metric and momentum generator
- [ ] `chmc` kernels

_Originally posted by @krzysztofrusek in https://github.com/blackjax-devs/blackjax/discussions/642_
